From 8785cfa9ce642d8b4cde64ea0e4c497aa55a5780 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 17 Nov 2014 20:15:52 +0100
Subject: [PATCH] init: lvm: run vgchange even if vgscan does not return
 success

It seems like vgscan does not return success in initramfs, even if
the needed devices noded are created.

Ignore errors and try run vgchange in any case.
---
 initramfs-init.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index 0ff8b6b..7831995 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -258,8 +258,8 @@ start_cryptsetup() {
 start_lvm() {
 	[ -x /sbin/lvm ] || return
 	modprobe dm-mod
-	lvm vgscan --mknodes --ignorelockingfailure >/dev/null 2>&1 &&\
-		lvm vgchange --ignorelockingfailure -a y >/dev/null 2>&1
+	lvm vgscan --mknodes --ignorelockingfailure >/dev/null 2>&1
+	lvm vgchange --ignorelockingfailure -a y >/dev/null 2>&1
 }
 
 # determine the default interface to use if ip=dhcp is set
-- 
2.1.3

